packages/@aws-cdk/cloudformation-diff/lib/diff/template-and-changeset-diff-merger.ts

```diff
@@ -0,0 +1,139 @@
+// The SDK is only used to reference `DescribeChangeSetOutput`, so the SDK is added as a devDependency.
+// The SDK should not make network calls here
+import type { DescribeChangeSetOutput as DescribeChangeSet, ResourceChangeDetail as RCD } from '@aws-sdk/client-cloudformation';
+import * as types from '../diff/types';
+
+export type DescribeChangeSetOutput = DescribeChangeSet;
+type ChangeSetResourceChangeDetail = RCD;
+
+/**
+ * The purpose of this class is to include differences from the ChangeSet to differences in the TemplateDiff.
+ */
+export class TemplateAndChangeSetDiffMerger {
+  // If we somehow cannot find the resourceType, then we'll mark it as UNKNOWN, so that can be seen in the diff.
+  static UNKNOWN_RESOURCE_TYPE = 'UNKNOWN_RESOURCE_TYPE';
+
+  public static determineChangeSetReplacementMode(propertyChange: ChangeSetResourceChangeDetail): types.ReplacementModes {
+    if (propertyChange.Target?.RequiresRecreation === undefined) {
+      // We can't determine if the resource will be replaced or not. That's what conditionally means.
+      return 'Conditionally';
+    }
+
+    if (propertyChange.Target.RequiresRecreation === 'Always') {
+      switch (propertyChange.Evaluation) {
+        case 'Static':
+          return 'Always';
+        case 'Dynamic':
+          // If Evaluation is 'Dynamic', then this may cause replacement, or it may not.
+          // see 'Replacement': https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_ResourceChange.html
+          return 'Conditionally';
+      }
+    }
+
+    return propertyChange.Target.RequiresRecreation as types.ReplacementModes;
+  }
+
+  changeSet: DescribeChangeSetOutput | undefined;
+  changeSetResources: types.ChangeSetResources;
+
+  constructor(
+    args: {
```

Comment:
Nit: The convention we typically go for is constructor arguments being passed in via a defined interface. Go a little deeper, we try to consolidate all required property on a props interface and all optional properties on an options interface. Typically the props interface would extends the options interface. Something like:

```ts
interface TemplateAndChangeSetDiffMergerOptions {
  /*
   * Description of property
   *
   * @default - description of default
  */
  readonly changeSetResources?: types.ChangeSetResources;
}
```

```ts
export interface TemplateAndChangeSetDiffMergerProps extends TemplateAndChangeSetDiffMergerOptions {
  /*
   * Description of property
  */
  readonly changeSet: DescribeChangeSetOutput;
}
```

Then,

```ts
constructor(props: TemplateAndChangeSetDiffMergerProps) {}
```

Reply:
great, thanks!